### PR TITLE
#trivial - f-contact preferences - Chrome/Visual (Percy) Tests

### PR DESCRIPTION
**Pull request by oliversweb**
### Added
- Added Chrome & Visual Tests.
---
Plus
* Updated the Story helper to have collection of api mocks per selected state.
* Setup percy project https://percy.io/4bc223d1/f-contact-preferences

### CODE REVIEW

File: packages/components/pages/f-contact-preferences/package.json [hunk 1]
```diff
@@ -32,7 +32,8 @@
     "lint": "vue-cli-service lint",
     "lint:fix": "yarn lint --fix",
     "lint:style": "vue-cli-service lint:style",
-    "test": "vue-cli-service test:unit"
```
Comment (LTurns) [inline packages/components/pages/f-contact-preferences/package.json:35]: just need to add the component script here too -  `"test-component:chrome": "cross-env-shell TEST_TYPE=component wdio ../../../../wdio-chrome.conf.js",`

File: packages/components/pages/f-contact-preferences/test-utils/component-objects/f-contactPreferences.component.js [hunk 2]
```diff
@@ -1,14 +1,24 @@
...
-    get component () {
-        return $('[data-test-id="contactPreferences"]');
-    }
+    get component () { return $(COMPONENT); }
+
+    get errorPage () { return $(ERROR_PAGE); }
+
+    get submitButton () { return $(SUBMIT_BUTTON); }
+
+    get newsEmailPreference () { return $(CHECKBOXES.news.email); }
```
Comment (siggerzz) [inline packages/components/pages/f-contact-preferences/test-utils/component-objects/f-contactPreferences.component.js:21]: Add new selector for newsSmsPreference

File: packages/components/pages/f-contact-preferences/test-utils/component-objects/f-contactPreferences.component.js [hunk 3]
```diff
@@ -17,4 +27,16 @@ module.exports = class ContactPreferences extends Page {
...
+
+    isErrorPageDisplayed () {
+        return this.errorPage.isDisplayed();
+    }
+
+    clickSubmitButton () {
+        return this.submitButton.click();
+    }
+
+    clickNewsEmailCheckbox () {
```
Comment (siggerzz) [inline packages/components/pages/f-contact-preferences/test-utils/component-objects/f-contactPreferences.component.js:41]: add `clickNewsSmsCheckbox` function

File: packages/components/pages/f-contact-preferences/test-utils/component-objects/f-contactPreferences.component.js (hunk 2 above)
Comment (siggerzz) [reply packages/components/pages/f-contact-preferences/test-utils/component-objects/f-contactPreferences.component.js:21]: Whilst you won't be utilising it here, it's good to have for any future CoreWeb tests 👍🏼

File: packages/components/pages/f-contact-preferences/test/visual/f-contact-preferences.visual.desktop.spec.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    ]).it('should display the %s error page', locale => {
+        // Arrange
+        contactPreferences
+            .withQuery('&knob-Locale', locale)
+            .withQuery('&knob-Set Api State', 'api-post-failed');
+        contactPreferences.load();
+        contactPreferences.waitForComponent();
+
+        // Act
+        contactPreferences.clickNewsEmailCheckbox(); // dirty the form to allow submit
```
Comment (siggerzz) [inline packages/components/pages/f-contact-preferences/test/visual/f-contact-preferences.visual.desktop.spec.js:51]: Use the `.withQuery('&knob-News - Email - Opted In', 'true')` function. Should be quicker. + it's consistent with the other test 👍🏼

File: packages/components/pages/f-contact-preferences/test/visual/f-contact-preferences.visual.mobile.spec.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    ]).it('should display the %s error page', locale => {
+        // Arrange
+        contactPreferences
+            .withQuery('&knob-Locale', locale)
+            .withQuery('&knob-Set Api State', 'api-post-failed');
+        contactPreferences.load();
+        contactPreferences.waitForComponent();
+
+        // Act
+        contactPreferences.clickNewsEmailCheckbox(); // dirty the form to allow submit
```
Comment (siggerzz) [inline packages/components/pages/f-contact-preferences/test/visual/f-contact-preferences.visual.mobile.spec.js:51]: ditto

**Review by siggerzz — CHANGES_REQUESTED** [review]
Looking great, just a few minor comments!

File: packages/components/pages/f-contact-preferences/test/visual/f-contact-preferences.visual.desktop.spec.js (hunk 4 above)
Comment (oliversweb) [reply packages/components/pages/f-contact-preferences/test/visual/f-contact-preferences.visual.desktop.spec.js:51]: @siggerzz Unsure that will raise the `isFormDirty` flag and if that does not get raised the Submit button will not trigger an error then the form will not show the error page

Comment (siggerzz) [reply packages/components/pages/f-contact-preferences/test/visual/f-contact-preferences.visual.desktop.spec.js:51]: No problem!

**Review by LTurns — DISMISSED** [review]

**Review by siggerzz — APPROVED** [review]

**Review by LTurns — APPROVED** [review]

**Pull request merged**